apps/entities/api/serializers.py

```diff
@@ -32,25 +26,18 @@ def get_serialized_end_date(self, instance: 'Entity'):
         return instance.end_date.serialize() if instance.end_date else None
 
 
-class EntitySerializer(SearchableModelSerializer):
+class EntityModelSerializer(TypedModuleSerializerDrf):
     """Serializer for entities."""
 
-    name = serpy.StrField()
-    slug = serpy.StrField()
-    unabbreviated_name = serpy.StrField()
-    aliases = serpy.Field()
-    birth_date = serpy.MethodField('get_serialized_birth_date')
-    death_date = serpy.MethodField('get_serialized_death_date')
-    description = serpy.StrField(required=False)
-    truncated_description = serpy.StrField()
-    cached_images = serpy.Field()
+    description = serializers.CharField(required=False)
     categorizations = CategorizationSerializer(
-        many=True, attr='categorizations.all', call=True
+        many=True, required=False, source='categorizations.all'
     )
 
-    def get_model(self, instance) -> str:  # noqa
-        """Return the model name of the instance."""
-        return f'entities.{instance.__class__.__name__.lower()}'
+    birth_date = serializers.DateField(write_only=True, required=False)
+    death_date = serializers.DateField(write_only=True, required=False)
+    birthDate = serializers.SerializerMethodField('get_serialized_birth_date', read_only=True)
```

Comment:
Do we need `birth_date` and `birthDate` (same with `death_date` and `deathDate`)?

In our `REST_FRAMEWORK` settings, we specify automatically converting between snake case and camel case, so ideally we should use snake case in our Python code and camel case in our JavaScript code.
```
'DEFAULT_RENDERER_CLASSES': (
        # https://github.com/vbabiy/djangorestframework-camel-case
        'djangorestframework_camel_case.render.CamelCaseJSONRenderer',
        'djangorestframework_camel_case.render.CamelCaseBrowsableAPIRenderer',
        'rest_framework.renderers.JSONRenderer',
    ),
    'DEFAULT_PARSER_CLASSES': (
        # https://github.com/vbabiy/djangorestframework-camel-case
        'djangorestframework_camel_case.parser.CamelCaseFormParser',
        'djangorestframework_camel_case.parser.CamelCaseMultiPartParser',
        'djangorestframework_camel_case.parser.CamelCaseJSONParser',
    ),
```

Reply:
Yes we need both, because input params and serialized version's aren't the same format.

Snake case version is the input and camel case version is serialized output, and they both will transformed to camel case version in the output. 
This way API client's only need to know about camel cased versions: `birthDate`, `deathDate` 
Isn't this the desired behavior? I we rename to `serialized_x_date`, the API clients will have to use `xDate` for inputs and `serializedXDate` when parsing outputs.

Reply:
I think what's confusing to me is that since `birth_date` and `death_date` are converted to `birthDate` and `deathDate` anyway before Django sends a response to the frontend, I'm not sure which value is received by the frontend. Will the `birthDate` value received by the frontend be the write-only `DateField` or the read-only `SerializerMethodField`?

Reply:
Since `birth_date/death_date` have `write_only=True`, they won't be rendered so frontend only receives serialized versions (`birthDate/deathDate` transformed to `birth_date/death_date` by DRF).

Reply:
Awesome, OK, that makes sense. Could you please add a comment to the code about this?